spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniparc/mapper/UniParcEntryJoin.java

```diff
@@ -0,0 +1,117 @@
+package org.uniprot.store.spark.indexer.uniparc.mapper;
+
+import static org.uniprot.core.uniparc.UniParcCrossReference.PROPERTY_SOURCES;
+import static org.uniprot.store.spark.indexer.uniparc.mapper.UniParcJoinUtils.*;
+
+import java.io.Serial;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.spark.api.java.Optional;
+import org.apache.spark.api.java.function.Function;
+import org.jetbrains.annotations.NotNull;
+import org.uniprot.core.Property;
+import org.uniprot.core.taxonomy.TaxonomyEntry;
+import org.uniprot.core.uniparc.UniParcCrossReference;
+import org.uniprot.core.uniparc.UniParcEntry;
+import org.uniprot.core.uniparc.impl.UniParcCrossReferenceBuilder;
+import org.uniprot.core.uniparc.impl.UniParcEntryBuilder;
+import org.uniprot.core.util.Utils;
+import org.uniprot.store.spark.indexer.uniparc.model.UniParcTaxonomySequenceSource;
+
+import lombok.extern.slf4j.Slf4j;
+import scala.Tuple2;
+
+/**
+ * @author lgonzales
+ * @since 21/01/2021
+ */
+@Slf4j
+public class UniParcEntryJoin
+        implements Function<
+                Tuple2<String, Tuple2<UniParcEntry, Optional<UniParcTaxonomySequenceSource>>>,
+                UniParcEntry> {
+
+    @Serial private static final long serialVersionUID = -6093868325315670369L;
+
+    @Override
+    public UniParcEntry call(
+            Tuple2<String, Tuple2<UniParcEntry, Optional<UniParcTaxonomySequenceSource>>> tuple)
+            throws Exception {
+        UniParcEntry result = tuple._2._1;
+        Optional<UniParcTaxonomySequenceSource> uniParcJoin = tuple._2._2;
```

Comment:
Please rename variable `uniParcJoin` to more meaningfulname. Can you please check all the names(methods, parameters, variable, classes) if they represent the intent of the code or not?